tutorials/dataframe/df106_HiggsToFourLeptons.C

```diff
@@ -0,0 +1,281 @@
+/// \file
+/// \ingroup tutorial_dataframe
+/// \notebook -draw
+/// The Higgs to four lepton analysis from the ATLAS Open Data release of 2020, with RDataFrame.
+///
+/// This tutorial is the Higgs to four lepton analysis from the ATLAS Open Data release in 2020
+/// (http://opendata.atlas.cern/release/2020/documentation/). The data was taken with the ATLAS detector
+/// during 2016 at a center-of-mass energy of 13 TeV. The decay of the Standard Model Higgs boson
+/// to two Z bosons and subsequently to four leptons is called the "golden channel". The selection leads
+/// to a narrow invariant mass peak on top a relatively smooth and small background, revealing
+/// the Higgs at 125 GeV.
+/// The analysis is translated to an RDataFrame workflow processing about 300 MB of simulated events and data.
+///
+/// Lepton scale factors are applied to leptons in MC simulations to correct for the differences in the
+/// reconstruction, identification and trigger mechanisms used for MC samples and real data.
+/// Systematic uncertainties for those MC scale factors are evaluated and the Vary function of RDataFrame is used to
+/// scale the variations to the final four leptons mass distribution.
+///
+///
+/// \macro_code
+/// \macro_image
+/// \date March 2020
+/// \author Stefan Wunsch (KIT, CERN)
+/// \date August 2022
+/// \author Julia Mathe (CERN)
+/// \date August 2023
+/// \author Marta Czurylo (CERN)
+
+#include <Math/Vector4D.h>
+#include <Math/Interpolator.h>
+#include <ROOT/RDFHelpers.hxx>
+#include <ROOT/RDataFrame.hxx>
+#include <ROOT/RVec.hxx>
+#include <TCanvas.h>
+#include <TH1D.h>
+#include <THStack.h>
+#include <TLatex.h>
+#include <TLegend.h>
+#include <TProfile.h>
+#include <TStyle.h>
+
+using namespace ROOT::VecOps;
+using ROOT::RVecF;
+using namespace ROOT::RDF::Experimental;
+
+// Define functions needed in the analysis
+// Select events for the analysis
+bool GoodElectronsAndMuons(const ROOT::RVecI &type, RVecF pt, RVecF eta, RVecF phi, RVecF e, RVecF trackd0pv,
+                           RVecF tracksigd0pv, RVecF z0)
+{
+   for (size_t i = 0; i < type.size(); i++) {
+      ROOT::Math::PtEtaPhiEVector p(0.001 * pt[i], eta[i], phi[i], 0.001 * e[i]);
+      if (type[i] == 11) {
+         if (pt[i] < 7000 || abs(eta[i]) > 2.47 || abs(trackd0pv[i] / tracksigd0pv[i]) > 5 ||
+             abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      } else {
+         if (abs(trackd0pv[i] / tracksigd0pv[i]) > 5 || abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      }
+   }
+   return true;
+}
+
+// Compute the invariant mass of a four-lepton-system.
+float ComputeInvariantMass(RVecF pt, RVecF eta, RVecF phi, RVecF e)
+{
+   ROOT::Math::PtEtaPhiEVector p1(pt[0], eta[0], phi[0], e[0]);
+   ROOT::Math::PtEtaPhiEVector p2(pt[1], eta[1], phi[1], e[1]);
+   ROOT::Math::PtEtaPhiEVector p3(pt[2], eta[2], phi[2], e[2]);
+   ROOT::Math::PtEtaPhiEVector p4(pt[3], eta[3], phi[3], e[3]);
```

Comment:
```suggestion
   using PtEtaPhiEVectorF = ROOT::Math::LorentzVector<ROOT::Math::PtEtaPhiE4D<float>>;
   PtEtaPhiEVectorF p1{pt[0], eta[0], phi[0], e[0]};
   PtEtaPhiEVectorF p2{pt[1], eta[1], phi[1], e[1]};
   PtEtaPhiEVectorF p3{pt[2], eta[2], phi[2], e[2]};
   PtEtaPhiEVectorF p4{pt[3], eta[3], phi[3], e[3]};
```

Your columns are in single precision, why use `PtEtaPhiEVector`, which is double precision? I don't think this is necessary. The users will copy-paste a lot from tutorials, and when they copy-paste this they will potentially waste many CPU hours!